### DISCUSSION TITLE
Add guide how to add validation and feedback

### DISCUSSION THREAD

**Discussion by benjamineckstein** [manojadams/mui-forms#116] [Q&A] [answered]
**Describe the solution you'd like**
I looked through your guide and got a basic idea how to use this library, but I miss information on how to add form validation on client side and handle validation errors from server side. 

- How would I add a validation for a field that a string needs to be at least 3 chars long? 
- How can I handle if the server returns a non 2xx code (e.g. 4xx) and provide some kind of error feedback to the user? 

To be more precise, there can be error messages related to one field (e.g. this field cannot be empty) or general messages why your form submit was not successful (e.g. if you select this checkbox, then you need to provide a title)

**Comment by manojadams** [comment, marked as answer]
@benjamineckstein Here are the docs:
- Custom field validation: https://mui-forms.vercel.app/fields/field-validation-custom
- Client side validation: https://mui-forms.vercel.app/fields/field-validation
- Server side validation: https://mui-forms.vercel.app/error-handling
